torch_struct/semirings/semirings.py

```diff
@@ -468,22 +469,94 @@ def prod(cls, xs, dim=-1):
     @classmethod
     def zero_mask_(cls, xs, mask):
         "Fill *ssize x ...* tensor with additive identity."
-        xs[0].masked_fill_(mask, -1e5)
-        xs[1].masked_fill_(mask, 0)
+        xs[0].masked_fill_(mask, cls.zero[0])
+        xs[1].masked_fill_(mask, cls.zero[1])
 
-    @staticmethod
-    def zero_(xs):
-        xs[0].fill_(-1e5)
-        xs[1].fill_(0)
+    @classmethod
+    def zero_(cls, xs):
+        xs[0].fill_(cls.zero[0])
+        xs[1].fill_(cls.zero[1])
         return xs
 
-    @staticmethod
-    def one_(xs):
-        xs[0].fill_(0)
-        xs[1].fill_(0)
+    @classmethod
+    def one_(cls, xs):
+        xs[0].fill_(cls.one[0])
+        xs[1].fill_(cls.one[1])
         return xs
 
 
+def ValueExpectationSemiring(k):
```

Comment:
Are you sure we don't have this already? Could have sworn someone added it.

Reply:
I'm not 100% sure, I looked and hadn't seen it anywhere in master so I went ahead with it. Maybe it's in another branch?